src/ProjectTemplates/ComponentsWebAssembly.ProjectTemplates/ComponentsWebAssembly-CSharp.Client.csproj.in

```diff
@@ -21,14 +21,10 @@
 
   <!--#endif -->
   <!--#if PWA -->
-  <ItemGroup>
-    <!-- When publishing, swap service-worker.published.js in place of service-worker.js -->
-    <Content Update="wwwroot\service-worker*.js" CopyToPublishDirectory="false" />
-    <ContentWithTargetPath Include="wwwroot\service-worker.published.js">
-      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
-      <TargetPath>wwwroot\service-worker.js</TargetPath>
-    </ContentWithTargetPath>
+  <!-- When publishing, swap service-worker.published.js in place of service-worker.js -->
+  <ItemGroup Condition="'$(DesignTimeBuild)' != 'true'">
+    <Content Remove="wwwroot\service-worker.js" />
+    <Content Update="wwwroot\service-worker.published.js" Link="wwwroot\service-worker.js" />
   </ItemGroup>
-
```

Comment:
I think this line is intentional. See similar pattern here: https://github.com/dotnet/aspnetcore/pull/18409/files#diff-6cf6f5bd480fbddd628e8bb2c9a5113fR21